fastai/dataloader.py

```diff
@@ -73,8 +73,8 @@ def np_collate(self, batch):
 
     def get_batch(self, indices):
         res = self.np_collate([self.dataset[i] for i in indices])
```

Comment:
res is itself a list of length equal to the number of outputs of __getitem__ method of DataSet. The old dataloader assumed this length to be 2 and transposed **only** the first array if transpose was set to True. The update makes the assumption that the last array in res list contains our target label, whereas all the preceding arrays are features.  It thus, contrary to the old dl, transposes all the arrays up to the last array in the res list if transpose is set to True.

Reply:
Thanks - so it works fine with DataLoader - but looks like it is indeed listifying the inputs, even if they're not originally lists. That isn't the expected behaviour of a DataSet, and might break some code.

Reply:
DataSets are not only used with DataLoader. They're also used directly. If you have just a single tensor as x, you're returning a list [x], rather than leaving it as-is. Does that make sense?

Reply:
Oh, I haven't made any changes to the DataSets, just the get_batch method of DataLoaders in case of transpose being set True. Also, the new dataloader doesn't return a list.
```
*x,y = next(iter(trn_dl))
```
The asterisk (*) is actually the reason why x ends up becoming a list. If you instead use the following, then each xi is a Tensor.
```
x1,....,xn, y = next(iter(trn_dl)) 
```